regcore/search/models.py

```diff
@@ -1,11 +1,41 @@
+from datetime import date
+
 from django.db import models
 
 from django.contrib.postgres.fields import ArrayField
-from django.contrib.postgres.search import SearchVector, SearchVectorField
+from django.contrib.postgres.search import (
+    SearchVector,
+    SearchVectorField,
+    SearchQuery,
+    SearchRank,
+    SearchHeadline,
+)
 
 from regcore.models import Part
 
 
+class SearchIndexQuerySet(models.QuerySet):
+    def search(self, query):
```

Comment:
There are two responsibilities in this function:
1. Query documents for a string in a consistent way
2. Limit that Query to only effective parts.

Separate the function into `effective()` and `search()` then we can say something like: `SearchIndex.objects.effective(date).search(query)`